src/dataset-builder/CohortEditor.ts

```diff
@@ -52,17 +52,28 @@ export const CriteriaView = (props: CriteriaViewProps) => {
 
   const [criteriaCount, setCriteriaCount] = useLoadedData<DatasetParticipantCountResponse>();
 
-  useEffect(() => {
-    setCriteriaCount(async () =>
-      DataRepo()
-        .dataset(datasetId)
-        .getCounts({
-          // Create a "cohort" to get the count of participants for this criteria on its own.
-          cohorts: [{ criteriaGroups: [{ criteria: [criteria], name: '', meetAll: true, mustMeet: true }], name: '' }],
-        })
-    );
-  }, [criteria, datasetId, setCriteriaCount]);
+  const debounceSetCriteriaCount = useRef(
+    _.debounce(250, (datasetId, criteria) => {
```

Comment:
Would it be possible to split out the `_.debounce(250, ...)` into a separate method so it could be shared in both places? Also it seems like we'd want a way to mock it during tests, so our tests wouldn't have to wait for the debounce to timeout before an expected text appears in the UI.

Reply:
If we were to do this, we could get rid of the useRef because useRefs are only allowed to be used in a component. Then, we can create an abstract function to get the debounce out. 

`const debounceFunc = _.debounce(250, (func) => func);`

Then, we would put the `func` inside of the `useEffect` like this: 


```
  useEffect(() => {
    debounceFunc(
      setCriteriaCount(
        async () =>
          await DataRepo()
            .dataset(datasetId)
            .getCounts({
              cohorts: [
                {
                  // Create a "cohort" to get the count of participants for this criteria on its own.
                  criteriaGroups: [{ criteria: [criteria], name: '', meetAll: true, mustMeet: true }],
                  name: '',
                },
              ],
            })
      )
    );
  }, [criteria, datasetId, setCriteriaCount]);
  
```

There could certainly be a stronger function name for `debounceFunc`, but I personally think this will clutter the `useEffect`? We can't set this function: 
```
      setCriteriaCount(
        async () =>
          await DataRepo()
            .dataset(datasetId)
            .getCounts({
              cohorts: [
                {
                  // Create a "cohort" to get the count of participants for this criteria on its own.
                  criteriaGroups: [{ criteria: [criteria], name: '', meetAll: true, mustMeet: true }],
                  name: '',
                },
              ],
            })
      )
```
as a variable outside of the `useEffect` because this is going to cause an infinite render error and we also can't put it inside of the `useEffect` because this will be undebounced.